As a user, I would like to be able to include custom time_pools with event detection

This is to satisfy the potential requirement of ensuring the events include historic events of significance that event detection could overlook. Per James in #382:

> For example, let's say that event detection fails to identify a historically important event and a user wants to add it in alongside the detected events. Sure, that indicates a limitation with event detection, but I guess a user should be able to mitigate this. Assuming it isn't additional work, I lean towards demoting this to a warning rather than an error. I don't think it makes sense to allow in combination with a regular sequence of valid_date_pools, however, and this should produce an error.

The exception that occurs when including `time_pools` with `event_detection` is this:

```
Caused by: wres.config.yaml.DeclarationException: Encountered 1 error(s) in the declared evaluation, which must be fixed:
    - Event detection was declared alongside explicit time pools, which is not allowed because event detection also generates time pools. Please remove the declaration of either 'event_detection' or 'time_pools' and try again.
```

The declaration I was using to test this capability was `.../wresTestData/github382/single_valued_ahps_declaration.value_restricted.time_pools.yml`. I don't think its necessary for me to share it to implement the feature, but I'm providing it here so that I can reuse it later to UAT the feature.  All I did was take the declaration in #404 and define a multi year time pool in the middle of the overall `reference_dates`. 

Thanks,

Hank

--- a/src/wres/pipeline/pooling/PoolFactory.java
+++ b/src/wres/pipeline/pooling/PoolFactory.java
@@ -497,25 +497,37 @@ public class PoolFactory
                                                                     Set<FeatureGroup> featureGroups,
                                                                     RetrieverFactory<Double, Double, Double> eventRetriever )
     {
-        // Time windows without event detection
-        if ( Objects.isNull( declaration.eventDetection() ) )
-        {
-            // Declared time windows
-            Set<TimeWindowOuter> timeWindows = TimeWindowSlicer.getTimeWindows( declaration );
+        // Declared time windows
+        Set<TimeWindowOuter> timeWindows = TimeWindowSlicer.getTimeWindows( declaration );
 
-            return featureGroups.stream()
-                                .collect( Collectors.toMap( Function.identity(),
-                                                            g -> timeWindows ) );
-        }
+        LOGGER.debug( "Generated {} declared time windows for evaluation, which will be used across all "
+                      + "geographic features.",
+                      timeWindows.size() );
 
-        // Time windows based on event detection
+        // Generate the feature-specific time-windows, including those associated with event detection, as needed
         Map<FeatureGroup, Set<TimeWindowOuter>> featurefulWindows = new HashMap<>();
         for ( FeatureGroup nextGroup : featureGroups )
         {
-            Set<TimeWindowOuter> events = this.getEventsGenerator()
-                                              .doEventDetection( project, nextGroup, eventRetriever );
+            Set<TimeWindowOuter> allTimeWindows = new TreeSet<>( timeWindows );
 
-            featurefulWindows.put( nextGroup, events );
+            // Event detection
+            if ( Objects.nonNull( declaration.eventDetection() ) )
+            {
+                Set<TimeWindowOuter> events = this.getEventsGenerator()
+                                                  .doEventDetection( project, nextGroup, eventRetriever );
+
+                allTimeWindows.addAll( events );
+
+                LOGGER.debug( "Generated {} time windows from event detection for the evaluation of feature group {}, "
+                              + "which will be added to the {} declared time windows, producing {} time windows in "
+                              + "total for this feature group.",
+                              events.size(),
+                              nextGroup,
+                              timeWindows.size(),
+                              allTimeWindows.size() );
+            }
+
+            featurefulWindows.put( nextGroup, Collections.unmodifiableSet( allTimeWindows ) );
         }
 
         return Collections.unmodifiableMap( featurefulWindows );

--- a/wres-config/src/wres/config/yaml/DeclarationValidator.java
+++ b/wres-config/src/wres/config/yaml/DeclarationValidator.java
@@ -2281,18 +2281,18 @@ public class DeclarationValidator
             events.add( error );
         }
 
-        // Error when also declaring explicit time pools
+        // Warn when also declaring explicit time pools, but allow because a user may want to guarantee an explicit pool
         if ( !declaration.timePools()
                          .isEmpty() )
         {
             EvaluationStatusEvent error
                     = EvaluationStatusEvent.newBuilder()
-                                           .setStatusLevel( StatusLevel.ERROR )
+                                           .setStatusLevel( StatusLevel.WARN )
                                            .setEventMessage( "Event detection was declared alongside explicit time "
-                                                             + "pools, which is not allowed because event detection "
-                                                             + "also generates time pools. Please remove the "
-                                                             + "declaration of either 'event_detection' or "
-                                                             + "'time_pools' and try again." )
+                                                             + "pools, which is allowed, but may not be intended as "
+                                                             + "event detection also generates time pools. If this is "
+                                                             + "not intended, please remove the declaration of either "
+                                                             + "'event_detection' or 'time_pools' and try again." )
                                            .build();
             events.add( error );
         }

--- a/wres-config/test/wres/config/yaml/DeclarationValidatorTest.java
+++ b/wres-config/test/wres/config/yaml/DeclarationValidatorTest.java
@@ -2660,8 +2660,8 @@ class DeclarationValidatorTest
                                                                         StatusLevel.ERROR ) ),
                    () -> assertTrue( DeclarationValidatorTest.contains( events,
                                                                         "Event detection was declared alongside explicit "
-                                                                        + "time pools, which is not allowed",
-                                                                        StatusLevel.ERROR ) ),
+                                                                        + "time pools, which is allowed",
+                                                                        StatusLevel.WARN ) ),
                    () -> assertTrue( DeclarationValidatorTest.contains( events,
                                                                         "Event detection was declared alongside lead time "
                                                                         + "pools",

--- a/wres-io/src/wres/io/ingesting/SourceLoader.java
+++ b/wres-io/src/wres/io/ingesting/SourceLoader.java
@@ -207,12 +207,12 @@ public class SourceLoader
 
     static <T> CompletableFuture<Object> doAllOrException( final List<CompletableFuture<T>> futures )
     {
-        //Complete when all futures are completed
+        // Complete when all futures are completed
         final CompletableFuture<Void> allDone =
                 CompletableFuture.allOf( futures.toArray( new CompletableFuture[0] ) );
-        //Complete when any of the underlying futures completes exceptionally
+        // Complete when any of the underlying futures completes exceptionally
         final CompletableFuture<T> oneExceptional = new CompletableFuture<>();
-        //Link the two
+        // Link the two
         for ( final CompletableFuture<T> completableFuture : futures )
         {
             //When one completes exceptionally, propagate
@@ -221,7 +221,7 @@ public class SourceLoader
                 return null;
             } );
         }
-        //Either all done OR one completes exceptionally
+        // Either all done OR one completes exceptionally
         return CompletableFuture.anyOf( allDone, oneExceptional );
     }
 
@@ -253,6 +253,7 @@ public class SourceLoader
         List<CompletableFuture<List<IngestResult>>> savingSources = new ArrayList<>();
         for ( DataSource source : sources )
         {
+            LOGGER.debug( "Loading source {}.", source );
             List<CompletableFuture<List<IngestResult>>> results = this.loadSource( source );
             savingSources.addAll( results );
         }

--- a/wres-reading/src/wres/reading/netcdf/nwm/NwmTimeSeries.java
+++ b/wres-reading/src/wres/reading/netcdf/nwm/NwmTimeSeries.java
@@ -169,7 +169,8 @@ class NwmTimeSeries implements Closeable
                          this.readExecutor,
                          abandoned,
                          ie );
-            Thread.currentThread().interrupt();
+            Thread.currentThread()
+                  .interrupt();
         }
 
         if ( !this.readExecutor.isShutdown() )
@@ -295,7 +296,8 @@ class NwmTimeSeries implements Closeable
             {
                 LOGGER.warn( "Interrupted while opening netCDF resources.", ie );
                 this.close();
-                Thread.currentThread().interrupt();
+                Thread.currentThread()
+                      .interrupt();
             }
             catch ( ExecutionException ee )
             {
@@ -1369,6 +1371,7 @@ class NwmTimeSeries implements Closeable
         @Override
         public NetcdfFile call() throws IOException
         {
+            LOGGER.debug( "Opening NetCDF resource, '{}'", this.uri );
             return NetcdfFiles.open( this.uri.toString() );
         }
     }
@@ -1912,7 +1915,8 @@ class NwmTimeSeries implements Closeable
                 //Read the features from the netCDF file and clone them to ensure that
                 //we are referring to a fresh copy, and not an internal netCDF copy.
                 LOGGER.debug( "Reading features from {}", netcdfFileName );
-                long[] features = ( long[] ) featureVariable.read().get1DJavaArray( DataType.LONG );
+                long[] features = ( long[] ) featureVariable.read()
+                                                            .get1DJavaArray( DataType.LONG );
                 if ( features == null )
                 {
                     throw new IllegalStateException( "netCDF library returned null array when looking for NWM features." );

